src/modules/job-exec/job-exec.c

```diff
@@ -171,12 +171,14 @@ flux_future_t *jobinfo_shell_rpc_pack (struct jobinfo *job,
     char *shell_topic = NULL;
     flux_future_t *f = NULL;
     uint32_t rank;
-
-    if (asprintf (&shell_topic,
-                  "%ju-shell-%s.%s",
-                  (uintmax_t) job->userid,
-                  idf58 (job->id),
-                  topic) < 0
+    char idbuf[21];
+
+    if (flux_job_id_encode (job->id, "f58plain", idbuf, sizeof (idbuf)) < 0
```

Comment:
Would it make sense to also have `idf58plain()` convenience function to simplify here and the shell's `build_topic()`?

Reply:
It felt to me like usage of this would be fairly limited, so probably not worth doing that?   Maybe later if that turns out to not be the case?